### PR TITLE
libbacktrace: fix build on Windows + resolve build requires conflicts

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I've also tried to build libbacktrace with Visual Studio. After fighting with a build context and autotools parameters, I was able to pass configure step but compilation fails due to pointer arithmetic:

```
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(158): warning C4820: 'debug_section_info': '4' bytes padding added after data member 'offset'
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(188): warning C4100: 'callback': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(187): warning C4100: 'pc': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(186): warning C4100: 'state': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(201): warning C4100: 'callback': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(200): warning C4100: 'addr': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(199): warning C4100: 'state': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(361): warning C4267: 'function': conversion from 'size_t' to 'uint16_t', possible loss of data
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(407): warning C4267: 'function': conversion from 'size_t' to 'uint16_t', possible loss of data
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(452): warning C4245: '=': conversion from 'int' to 'uintptr_t', signed/unsigned mismatch
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(531): warning C4100: 'error_callback': unreferenced formal parameter
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(651): error C2036: 'const void *': unknown size
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(685): error C2036: 'const void *': unknown size
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(729): error C2036: 'const void *': unknown size
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(731): warning C4267: '=': conversion from 'size_t' to 'off_t', possible loss of data
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(750): warning C4456: declaration of 'str_off' hides previous local declaration
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(599): note: see declaration of 'str_off'
C:/users/spaceim/.conan/data/libbacktrace/cci.20210118/_/_/build/3fb49604f9c2f729b85ba3115852006824e72cab/source_subfolder/pecoff.c(825): warning C4267: '=': conversion from 'size_t' to 'off_t', possible loss of data
```

https://github.com/ianlancetaylor/libbacktrace/blob/dedbe13fda00253fe5d4f2fb812c909729ed5937/pecoff.c#L651
https://github.com/ianlancetaylor/libbacktrace/blob/dedbe13fda00253fe5d4f2fb812c909729ed5937/pecoff.c#L685
https://github.com/ianlancetaylor/libbacktrace/blob/dedbe13fda00253fe5d4f2fb812c909729ed5937/pecoff.c#L729

**Review by madebr — DISMISSED** [review]

**Comment by SpaceIm** [comment]
I have a patch to build libbacktrace with Visual Studio, but I don't know if it makes sense since unwind is required at runtime.

File: recipes/libbacktrace/all/patches/0002-msvc-unistd-alternative.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
+--- a/allocfail.c
```
Comment (madebr) [inline recipes/libbacktrace/all/patches/0002-msvc-unistd-alternative.patch:1]: Do you intend to upstream this patch?

**Review by madebr — APPROVED** [review]

File: recipes/libbacktrace/all/patches/0002-msvc-unistd-alternative.patch (hunk 1 above)
Comment (SpaceIm) [reply recipes/libbacktrace/all/patches/0002-msvc-unistd-alternative.patch:1]: Not sure. I don't know if it really makes sense to support Visual Studio. I didn't fix all the files actually, only the ones built when libunwind is not found.

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**